doc/TockBinaryFormat.md

```diff
@@ -293,6 +309,35 @@ a non-position-independent binary at an incorrect location.
     the linker. If a fixed address is not required this should be set to
     `0xFFFFFFFF`.
 
+#### `6` Permissions
+
+The `Permissions` section allows an app to specify driver permissions that it
+is allowed to use. All driver syscalls that an app will use must be listed. The
+list should not include drivers that are not being used by the app.
+
+The data is stored in the optional `TbfHeaderV2Permissions` field. This
+includes an array of all the `perms`.
+
+```
+0             2             4
++-------------+-------------+---------...--+
+| Type (6)    | Length      | perms        |
++-------------+-------------+---------...--+
+```
+
+The `perms` array is made up of a number of elements of
+`TbfHeaderDriverPermission`. The length of the TLV can be used to determine
+the number of array elements. The elements in `TbfHeaderDriverPermission` are
+described below:
+
+`driver_number` is the number of the driver that is allowed. This for example
+could be `0x00000` to indicate that the `Alarm` syscalls are allowed.
+
+`allowed_commands` is a bit mask of the allowed commands. For example a value
+of `0b0001` indicates that only command 0 is allowed. `0b0111` would indicate
+that commands 2, 1 and 0 are all allowed. Subscribe and allow commands are
```

Comment:
Should `subscribe` be mass-permitted? One might imagine an information leakage channel coming from events being reported. I believe that in many cases right now, if Application A tells something to start sampling, an Application B can subscribe to the event stream without explicitly starting the sampling itself.

Reply:
I thought application B shouldn't get an event callback unless it started the command.

Reply:
Also, I'll clarify this that `allow` and `subscribe` are only permitted if the driver is permitted. If the driver is permitted all `allow` and `subscribe` calls are allowed.

Reply:
I think that clarification is good. However, we do have implementations that don't enforce that right now, e.g. for the ADC:

```rust
            // perform callback
            self.appid.map(|id| {
                self.apps
                    .enter(*id, |app, _| {
                        app.callback.map(|callback| {
                            calledback = true;
                            callback.schedule(
                                AdcMode::ContinuousSample as usize,
                                self.channel.get(),
                                sample as usize,
                            );
                        });
                    })
```

On the upcall path, callbacks will be issued to any apps that have subscribed, regardless of which app started the ADC. That may be a bug / unintended consequence in the ADC interface, but it seems sensible that if we're going to filter defensively, we should filter against apps "listening-in" when they didn't declare their intent/ability to do so.